qcodes_contrib_drivers/drivers/SingleQuantum/SingleQuantum.py

```diff
@@ -0,0 +1,498 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2020 Single Quantum B. V. and Andreas Fognini
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+import numpy as np
+import socket
+import json
+import threading
+import time
+import sys
+
+from qcodes.instrument.ip import IPInstrument
+from qcodes.instrument.base import Parameter
+from qcodes.instrument.parameter import ParameterWithSetpoints, MultiParameter
+from qcodes.utils.validators import Arrays
+
+
+def synchronized_method(method, *args, **kws):
+    outer_lock = threading.Lock()
+    lock_name = "__" + method.__name__ + "_lock" + "__"
+
+    def sync_method(self, *args, **kws):
+        with outer_lock:
+            if not hasattr(self, lock_name):
+                setattr(self, lock_name, threading.Lock())
+            lock = getattr(self, lock_name)
+            with lock:
+                return method(self, *args, **kws)
+    sync_method.__name__ = method.__name__
+    sync_method.__doc__ = method.__doc__
+    sync_method.__module__ = method.__module__
+    return sync_method
+
+
+def synchronized_with_attr(lock_name):
```

Comment:
It this ever used? I cannot find it while searching

Reply:
This part was legacy code, I removed and tested it.
Everything still works.